src/main/java/com/capitalone/dashboard/collector/SonarCollectorTask.java

```diff
@@ -248,69 +247,76 @@ private void deleteUnwantedJobs(List<SonarProject> latestProjects, List<SonarPro
     private void refreshData(List<SonarProject> sonarProjects, SonarClient sonarClient) {
         long start = System.currentTimeMillis();
         int count = 0;
-
         for (SonarProject project : sonarProjects) {
-            CodeQuality codeQuality = sonarClient.currentCodeQuality(project);
-            if (codeQuality != null && isNewQualityData(project, codeQuality)) {
-                project.setLastUpdated(System.currentTimeMillis());
-                sonarProjectRepository.save(project);
-                codeQuality.setCollectorItemId(project.getId());
-                codeQualityRepository.save(codeQuality);
-                count++;
+            try {
+                CodeQuality codeQuality = sonarClient.currentCodeQuality(project);
+                if (codeQuality != null && isNewQualityData(project, codeQuality)) {
+                    project.setLastUpdated(System.currentTimeMillis());
+                    sonarProjectRepository.save(project);
+                    codeQuality.setCollectorItemId(project.getId());
+                    codeQualityRepository.save(codeQuality);
+                    count++;
+                }
+            } catch (HttpClientErrorException e) {
+                HttpStatus status = e.getStatusCode();
+                if (status.toString().equals("404")) {
+                    project.setEnabled(false);
+                    sonarProjectRepository.save(project);
+                }
             }
         }
         log("Updated", start, count);
```

Comment:
Log the number of project disabled here as well. Use splunk friendly format (name=value), such as "refreshData finished, total=1234, updated=234, disables=45, timeTaken=2333"